### PR TITLE
feat: add listview

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: jina/types/list.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        """
+        Insert any proto object into the list
+
+        :param index: Position of the insertion.
+        :param object: The object that needs to be inserted.
+        """
+        self._pb_body.insert(index, object)
+
+    def __getitem__(self, i: Union[int, slice]):
+        if i < len(self):
```
Comment (nan-wang) [inline jina/types/list.py:34]: Raise an error when the index is out of range.
```suggestion
        if i >= len(self):
            raise IndexError('list index out of range')
        ...
```

File: jina/types/list.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+from collections.abc import MutableSequence
+from typing import Union
+
+from google.protobuf import struct_pb2
+
+from .mixin import ProtoTypeMixin
+
+
+class ListView(ProtoTypeMixin, MutableSequence):
```
Comment (nan-wang) [inline jina/types/list.py:9]: We also need `__eq__`.

```python
def test_list_view_error(list_proto):
    view_1 = ListView(list_proto)
    view_2 = ListView(list_proto)
    assert view_2 == view_1
```

**Review by nan-wang — CHANGES_REQUESTED** [review]
good job!

File: jina/types/list.py (hunk 1 above)
Comment (JoanFM) [reply jina/types/list.py]: Done!

File: jina/types/list.py (hunk 2 above)
Comment (JoanFM) [reply jina/types/list.py:9]: Done!

**Review by KingTropea999 — APPROVED** [review]

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Review by deepankarm — DISMISSED** [review]

File: jina/types/list.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    """
+
+    def __init__(self, list: struct_pb2.ListValue):
+        """Create a Python list view of Protobuf ListValue.
+
+        :param list: the protobuf ListValue object
+        """
+        self._pb_body = list
+
+    def insert(self, index: int, object) -> None:
```
Comment (bwanglzu) [inline jina/types/list.py:24]: what is the type of a object?

File: jina/types/list.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    def __eq__(self, other: Union['ListView', List]):
+        if isinstance(other, ListView):
+            return list(self) == list(other)
+        elif isinstance(other, List):
+            return list(self) == other
+        else:
+            return False
+
+    def __contains__(self, item):
```
Comment (bwanglzu) [inline jina/types/list.py:67]: maybe we should unify these namings? such as `item`, `value`, `object` etc..

**Review by bwanglzu — COMMENTED** [review]
left some comments

File: jina/types/list.py (hunk 3 above)
Comment (JoanFM) [reply jina/types/list.py:24]: MutableSequence interface

File: jina/types/list.py (hunk 4 above)
Comment (JoanFM) [reply jina/types/list.py]: Done

**Comment by JoanFM** [comment]
> left some comments

Taken care of. Mantained `MutableSequence` naming and changed `object` for `contains`

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**